src/optimizer/dataclients/png.rs

```diff
@@ -0,0 +1,28 @@
+extern crate image;
+extern crate dssim;
+extern crate imgref;
+extern crate lodepng;
+
+use std::env;
+use std::io;
+use std::path::{Path};
+use image::{open, ImageResult, DynamicImage};
+use imgref::{ImgVec};
+use dssim::*;
+use dataclients::ImageSpec;
+
+#[derive(Debug)]
+pub struct Png {
+    name: String
+}
+
+impl ImageSpec for Png {
+    fn new(&self, name: String) -> Png {
+        Png{ name }
```

Comment:
what is the usage of name here?

Reply:
Name could be the unique identifier for the specific trait implementation which can be used for logging and debugging purposes later